### PR TITLE
Lab adapter

### PR DESCRIPTION
**Pull request by MichalOgloza**
Zadania wykonane do punktu 3.4.

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java [hunk 1]
```diff
@@ -4,15 +4,22 @@
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class LineDrawerAdapter extends DrawPanelController implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:12]: Inheritance is not used (no super drawLine calls). Should be removed!
The name is better but it lacks a driver aspect.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java [hunk 2]
```diff
@@ -23,11 +30,12 @@ public void setPosition(int x, int y) {
...
 	@Override
 	public void operateTo(int x, int y) {
-		ILine line = LineFactory.getBasicLine();
 		line.setStartCoordinates(this.startX, this.startY);
 		line.setEndCoordinates(x, y);
+		this.startX = x;
+		this.startY = y;
 
-		drawLine(line);
+		DrawerFeature.getDrawerController().drawLine(line);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:38]: Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java [hunk 3]
```diff
@@ -23,11 +30,12 @@ public void setPosition(int x, int y) {
 
 	@Override
 	public void operateTo(int x, int y) {
-		ILine line = LineFactory.getBasicLine();
 		line.setStartCoordinates(this.startX, this.startY);
 		line.setEndCoordinates(x, y);
+		this.startX = x;
+		this.startY = y;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:36]: minor code duplication

**Review by kowallus — COMMENTED** [review]
4.5 pts
UML diagram does not show the relation with DrawerFeature.
Incorrect answer (the question concerned exclusively the CLASS variant of the adapter pattern).

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/ModifiableLineAdapter.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    public void setThickness(float thickness)
+    {
+        this.thickness = thickness;
+    }
+
+    @Override
+    public void setDotted(boolean dotted)
+    {
+        this.dotted = dotted;
+    }
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/ModifiableLineAdapter.java:26]: is it tested anywhere?!

Comment (MichalOgloza) [reply src/main/java/edu/kis/powp/jobs2d/drivers/adapter/ModifiableLineAdapter.java:26]: Yes, in TestJobs2dPatterns class.

Comment (kowallus) [reply src/main/java/edu/kis/powp/jobs2d/drivers/adapter/ModifiableLineAdapter.java:26]: No. This is the last commit in PR.

Comment (MichalOgloza) [reply src/main/java/edu/kis/powp/jobs2d/drivers/adapter/ModifiableLineAdapter.java:26]: Right, i forgot to push my last changes. Can i gain some extra points by preparing another PR with some issues fixed ?

Comment (kowallus) [reply src/main/java/edu/kis/powp/jobs2d/drivers/adapter/ModifiableLineAdapter.java:26]: Yes, but not many.

**Pull request closed without merging**